### ISSUE TITLE
Not working on ultrawide monitor

### ISSUE THREAD

**Issue by twkrol** [nexxyz/WinPan#1]
Hi, tried your tool on my setup with ultrawide monitor 5120x1440 configured as one huge monitor, not two or three.
Unfortunatelly, it doesn't detect movement of my Edge window playing twitch, the sound is always central.

**Comment by nexxyz** [comment]
Interesting. The type of monitor should not matter. I suspect that the
audio stream detection is for some reason not triggering with your
Edge/Twitch stream. Do you have other programs with potential (or actual)
audio output open?

**Comment by twkrol** [comment]
You're right: with spotify it works good, but with web browsers it doesn't: either Brave or Edge.
So probably the detection procedure could be improved.
BTW: when the app with audio is closed the balance stay there, but for better experience it should be reverted to central, otherwise, when starting WinPan it restores the last balance settings, event if the sounding app is not running.

**Comment by nexxyz** [comment]
Thanks for the feedback!

Yeah it could be improved for sure ☺️ it works for my own use-case right
now, so I’m not sure I will get to improving it, but I might. Also if you
decide to work on it, let me know!

It should actually restore the original balance when you exit WinPan. I
need to check if it should restore original balance when no stream is
detected anymore, that might be quick to do.

**Comment by twkrol** [comment]
Maybe the following post would be helpfull:
https://stackoverflow.com/questions/14306048/controlling-volume-mixer

**Comment by nexxyz** [comment]
Yeah I am using similar code. However I made it independent of the
application name, just triggering when only a single app is providing
audio. So as long as only one audio source is registered, it pans it where
the window is located relative to the center.

The Windows API does not allow panning an individual app‘s audio from the
outside (it only allows overall volume control to individual apps, but not
panning), so WinPan had to use the „global“ pan. That also means that it
just does not make a lot of sense to pan stuff around if multiple audio
sources are active.

**Comment by twkrol** [comment]
I see.
regarding the mixer: It looks like an app appears there if it starts playing audio once, and keep being mixed even when audio is not playing anymore. For example: when you open YouTube on a browser card the browser show in mixer, when you stop it stays there (don't know for how long)

**Comment by twkrol** [comment]
myabe an idea could be to get all apps present in mixer, but pan only to the one is active - for best results when is actively playing sound but how to find when a browser is playing sound?

**Comment by nexxyz** [comment]
Yeah. I think there are a couple of ways. Another could be to specify a
process/executable name on the command line as an optional argument, and
then to detect that one window and pan the whole thing accordingly.

**Comment by KadirFiratFTW** [comment]
Hi,

I run the app ass administrator and just tried with Spotify and there is nothing changed. Sound always centralized. 

OS: Win10

**Comment by KadirFiratFTW** [comment]
Btw i checked the source code and debugged it. My problem is related to channels. All the configuration is stereo but when the code is trying to get channel count and channels with "device.AudioEndpointVolume" it returns always 1. 

Stereo is working on my PC but WinPan.
